lib/Data/Validate/Sanctions.pm

```diff
@@ -208,30 +208,23 @@ sub get_sanctioned_info {    ## no critic (RequireArgUnpacking)
 
     $self->_load_data();
 
-    # Sub to remove non-alphabets from the name
-    my $clean_names = sub {
-
-        my ($full_name) = @_;
-
-        # Remove non-alphabets
-        my @cleaned_full_name = split " ", uc($full_name =~ s/[^[:alpha:]\s]//gr);
-
-        return @cleaned_full_name;
-    };
-
     my $client_full_name = join(' ', $first_name, $last_name || ());
 
     # Split into tokens after cleaning
-    my @client_name_tokens = $clean_names->($client_full_name);
+    my @client_name_tokens = _clean_names($client_full_name);
 
     my @match_with_dob_text;
 
-    my %index = $self->{_index}->%*;
-    foreach my $sanctioned_name (sort keys %index) {
-        my @sanctioned_name_tokens = $clean_names->($sanctioned_name);
-        next unless _name_matches(\@client_name_tokens, \@sanctioned_name_tokens);
+    my @sanctioned_names = ();
+    foreach my $token (@client_name_tokens) {
+        push(@sanctioned_names,  keys %{$self->{_token_sanctioned_names}->{$token}});
+    }
+
+    foreach my $sanctioned_name (sort @sanctioned_names) {
```

Comment:
Here we have a `sort` purely for the sake of passing tests which needs to change.
Sort is still expensive here and we should remove it.

Reply:
fixed here https://github.com/binary-com/perl-Data-Validate-Sanctions/pull/43